docs/keto/modeling/create-permission-model.mdx

```diff
@@ -0,0 +1,377 @@
+- The owner of a `document` is also an editor of that `document`.
+- An editor of a `document` is also a viewer of that `document`.
+- `Documents` can be arranged into a hierarchy of `folders`.
+- `Users` that can view the parent `folder` in the hierarchy can also view all `folders` and `documents` the parent `folder`
+  contains.
+
+After highlighting, you can identify the subject, `User`, and two objects: `Document` and `Folder`.
+
+This list translates into the first version of the permission model.
+
+```ts title="permissions-v1.ts"
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+// highlight-start
+class User implements Namespace {}
+class Document implements Namespace {}
+class Folder implements Namespace {}
+// highlight-end
+```
+
+:::tip
+
+In Ory Permissions, objects and subjects you identified are [namespaces](../concepts/namespaces.mdx). They are declared as classes
+in the Ory Permission Language. The convention is to name these classes like TypeScript classes, starting with a capital letter
+and and in the singular form, for example `Document` instead of `documents`.
+
+:::
+
+### Determine relationships between objects
+
+Read through the initial assumptions point by point to determine the list of relationships:
+
+- The goal is to model `user` access to `documents` such that `users` can be the `owner`, `editor`, or `viewer` of a `document`.
+
+  ```mdx-code-block
+  <Mermaid
+    chart={`
+  graph LR
+      Document -->|owner| User
+      Document -->|editor| User
+      Document -->|viewer| User
+  `}
+  />
+  ```
+
+- Every owner is also an editor of a `document` and every editor is also a viewer of a `document`. Furthermore, `documents` can be
+  put into a hierarchy of `folders`. If a user can view the parent `folder`, they can also view all `folders` and `documents` the
+  parent `folder` contains.
+
+  ```mdx-code-block
+  <Mermaid
+    chart={`
+  graph LR
+      Document -->|parent| Folder
+      Folder -->|parent| Folder
+      Folder -->|owner| User
+      Folder -->|editor| User
+      Folder -->|viewer| User
+  `}
+  />
+  ```
+
+This is the complete matrix of relationships presented on a single diagram:
+
+```mdx-code-block
+<Mermaid
+  chart={`
+graph LR
+    Document -->|owner| User
+    Document -->|editor| User
+    Document -->|viewer| User
+    Document -->|parent| Folder
+    Folder -->|parent| Folder
+    Folder -->|owner| User
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+```
+
+### Define relation tuples in the OPL
+
+Next, add each relation to the permission config.
+
+:::tip
+
+In Ory Permissions, relationships are declared inside the corresponding class in the Ory Permission Language. Since you store
+arbitrary tuples in Ory Permissions, you always have many-to-many relations between these objects. To reflect this in OPL, the
+convention is to pluralize the relation name, for example, `viewers` instead of `viewer`.
```

Comment:
I'm not 100% sure I understand `Since you store
arbitrary tuples in Ory Permissions, you always have many-to-many relations between these objects.` 
I think I know what you mean, but... Can you think of a simpler way to say this? 🤔

Reply:
I can't think of a much simpler way

Reply:
```suggestion
In Ory Permissions, relationships are declared inside the corresponding class in the Ory Permission Language. Ory Permissions only
has [many-to-many relationships](<https://en.wikipedia.org/wiki/Many-to-many_(data_model)>) between objects and subjects. To
reflect this in OPL, pluralize the relation name, for example, `viewers` instead of `viewer`.
```

WDYT?